x-pack/plugins/security_solution/public/management/pages/policy/store/policy_details/selectors/trusted_apps_selectors.test.ts

```diff
@@ -0,0 +1,405 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { PolicyDetailsState } from '../../../types';
+import { initialPolicyDetailsState } from '../reducer/initial_policy_details_state';
+import {
+  getCurrentArtifactsLocation,
+  getAvailableArtifactsList,
+  getAvailableArtifactsListIsLoading,
+  getUpdateArtifactsIsLoading,
+  getUpdateArtifactsIsFailed,
+  getUpdateArtifactsLoaded,
+  getAvailableArtifactsListExist,
+  getAvailableArtifactsListExistIsLoading,
+  getUpdateArtifacts,
+  isOnPolicyTrustedAppsPage,
+} from './trusted_apps_selectors';
+
+import { ImmutableObject } from '../../../../../../../common/endpoint/types';
+import {
+  createLoadedResourceState,
+  createUninitialisedResourceState,
+  createLoadingResourceState,
+  createFailedResourceState,
+} from '../../../../../state';
+import { MANAGEMENT_ROUTING_POLICY_DETAILS_TRUSTED_APPS_PATH } from '../../../../../common/constants';
+import { getListResponse, getAPIError, getFakeCreateResponse } from '../../../test_utils';
+
+describe('policy trusted apps selectors', () => {
+  let initialState: ImmutableObject<PolicyDetailsState>;
+
+  beforeEach(() => {
+    initialState = initialPolicyDetailsState();
+  });
+
+  describe('isOnPolicyTrustedAppsPage()', () => {
+    it('when location is on policy trusted apps page', () => {
+      const isOnPage = isOnPolicyTrustedAppsPage({
+        ...initialState,
+        location: {
+          pathname: MANAGEMENT_ROUTING_POLICY_DETAILS_TRUSTED_APPS_PATH,
+          search: '',
+          hash: '',
+        },
+      });
+      expect(isOnPage).toBeFalsy();
+    });
+    it('when location is not on policy trusted apps page', () => {
+      const isOnPage = isOnPolicyTrustedAppsPage({
+        ...initialState,
+        location: { pathname: '', search: '', hash: '' },
+      });
+      expect(isOnPage).toBeFalsy();
+    });
+  });
+
+  describe('getCurrentArtifactsLocation()', () => {
```

Comment:
I also see all these selectors tests as testing implementation details of redux. Considering selectors are function that filter the same information that is passed to them without any business logic.

Reply:
Same, just ensuring if some selector change, then the test will fail and will need to be changed

Reply:
I understand the logic behind them, I'm not saying you should remove them, they are already here after all, but if any selector would change, the tests for the react components and middleware would fail because they depend on them.

Reply:
Yes agree, but from my point of view, whatever is tested in his specific unit test, don't need to be tested in other test areas. So, react components tests will check the view is displayed correctly but not will check if some specific field provided by a selector has the properly value, I think this needs to be tested by a selector test as it can be used in react components but also in middleware or other code areas. Just my point of view :)